src/hooks/useFetch/index.ts

```diff
@@ -0,0 +1,60 @@
+import { useEffect, useState } from "react";
+import { isProduction } from "utils/environmentVariables";
+import { leaveBreadcrumb } from "utils/errorReporting";
+/**
+ * `useFetch` is a custom hook that downloads json from a given URL.
+ * It returns an object with the following properties:
+ * - isLoading: a boolean that is true while the log is being downloaded
+ * - data: the log file as an array of strings
+ * - error: an error message if the download fails
+ *
+ */
+const useFetch = <T extends object>(
+  url: string,
+  options?: {
+    skip?: boolean;
+  }
+) => {
+  const [data, setData] = useState<T | null>(null);
+  const [error, setError] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+
+  useEffect(() => {
+    leaveBreadcrumb("useFetch", { url }, "request");
+    const req = new Request(url, { method: "GET" });
+    const abortController = new AbortController();
+
+    if (!options?.skip) {
```

Comment:
it might be a little nicer if we deconstruct `skip` above
```suggestion
  const [isLoading, setIsLoading] = useState(false);
  const { skip = false } = options || {};

  useEffect(() => {
    leaveBreadcrumb("useFetch", { url }, "request");
    const req = new Request(url, { method: "GET" });
    const abortController = new AbortController();

    if (!skip) {
```